File: rust/examples/list_labels.py
"""Imprime labels por celda. Espejo de list_labels.rs."""
import sys

import gdstk


def main() -> None:
    if len(sys.argv) < 2:
        print("usage: list_labels.py <file.gds>", file=sys.stderr)
        sys.exit(2)

    lib = gdstk.read_gds(sys.argv[1])
    for cell in lib.cells:
        if not cell.labels:
            continue
        print(f"Cell '{cell.name}':")
        for lbl in cell.labels:
            x, y = lbl.origin
            print(
                f"  Layer {lbl.layer}/{lbl.texttype}  "
                f"({x:.2f}, {y:.2f})  '{lbl.text}'"
            )


if __name__ == "__main__":
    main()

File: rust/examples/list_labels.rs
//! Imprime labels por celda. Test de paridad para Fase 3.

use std::env;

use gdstk_rs::Library;

fn main() {
    let mut args = env::args();
    let program = args.next().unwrap_or_else(|| "list_labels".into());
    let Some(path) = args.next() else {
        eprintln!("usage: {program} <file.gds>");
        std::process::exit(2);
    };

    let lib = Library::open(&path);
    for cell in lib.cells() {
        if cell.label_count() == 0 {
            continue;
        }
        println!("Cell '{}':", cell.name());
        for label in cell.labels() {
            let p = label.origin();
            println!(
                "  Layer {}/{}  ({:.2}, {:.2})  '{}'",
                label.layer(),
                label.texttype(),
                p.x,
                p.y,
                label.text()
            );
        }
    }
}
